packages/api/amplify_api_dart/lib/src/graphql/web_socket/blocs/web_socket_bloc.dart

```diff
@@ -121,7 +121,7 @@ class WebSocketBloc with AWSDebuggable, AmplifyLoggerMixin {
     final subBloc = _saveRequest<T>(event);
 
     // Add subscribe event to [WebSocketBloc]
-    add(RegistrationEvent(event.request));
+    _registration(RegistrationEvent(event.request));
```

Comment:
nit: Could we change the signature of `_registration` (maybe rename `registerSubscriptionRequest`) to not take an event and just take the request itself? That way we could remove the `RegistrationEvent` class definition as it's no longer one of the events for the state machine, just a method used internally now.